x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/rule_editor_panel/json_rule_editor.test.tsx

```diff
@@ -12,25 +12,27 @@ import 'brace/mode/json';
 // warnings in the console which adds unnecessary noise to the test output.
 import '@kbn/test/target_node/jest/utils/stub_web_worker';
 
-import { EuiCodeEditor } from '@elastic/eui';
 import React from 'react';
 import { act } from 'react-dom/test-utils';
 
-import { mountWithIntl } from '@kbn/test/jest';
-import { coreMock } from 'src/core/public/mocks';
-import { KibanaContextProvider } from 'src/plugins/kibana_react/public';
+import type { monaco } from '@kbn/monaco';
+import { shallowWithIntl } from '@kbn/test/jest';
+import { CodeEditor } from 'src/plugins/kibana_react/public';
 
 import { AllRule, AnyRule, ExceptAllRule, ExceptAnyRule, FieldRule } from '../../model';
 import { JSONRuleEditor } from './json_rule_editor';
 
+jest.mock('../../../../../../../../src/plugins/kibana_react/public', () => ({
+  ...jest.requireActual('../../../../../../../../src/plugins/kibana_react/public'),
+  useKibana: jest.fn().mockReturnValue({
+    services: { docLinks: { links: { apis: { createRoleMapping: 'createRoleMappingLink' } } } },
+  }),
+}));
+
 describe('JSONRuleEditor', () => {
+  const mockChangeEvent = {} as monaco.editor.IModelContentChangedEvent;
   const renderView = (props: React.ComponentProps<typeof JSONRuleEditor>) => {
-    const coreStart = coreMock.createStart();
-    return mountWithIntl(
-      <KibanaContextProvider services={coreStart}>
-        <JSONRuleEditor {...props} />
-      </KibanaContextProvider>
-    );
+    return shallowWithIntl(<JSONRuleEditor {...props} />);
   };
```

Comment:
Jest uses jsdom to render components, but Monaco requires a canvas. Instead of installing a separate package just to use for this test, I opted for a shallow render which still allows us to exercise these unit test cases.

Note that to use a shallow mock this has to be the top level component, so as a result I had to mock `useKibana()`.